File: gmall-publisher/src/main/scala/com/atguigu/gmall/publisher/controller/PublisherController.scala
package com.atguigu.gmall.publisher.controller

import com.alibaba.fastjson.{JSON, JSONArray, JSONObject}
import com.atguigu.gmall.publisher.service.PublisherService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.{GetMapping, RestController}

import scala.collection.mutable
import scala.collection.mutable.ListBuffer


@RestController
class PublisherController @Autowired()(val publisherService: PublisherService){


  @GetMapping(Array("/realtime-total"))
  def getRealTimeTotal(date: String) = {
    val totalDau: Long = publisherService.getDauCountByDate(date)
    /** 操他妈 scala不支持序列化 依赖又下部下来jackson的scala */
    val jSONArray: JSONArray = new JSONArray()
    val object1: JSONObject = new JSONObject()
    object1.put("id","dau")
    object1.put("name","新增日活")
    object1.put("value",totalDau)
    jSONArray.add(object1)
    val object2: JSONObject = new JSONObject()
    object2.put("id","new_mid")
    object2.put("name","新增设备")
    object2.put("value",323)
    jSONArray.add(object2)

    jSONArray.toJSONString
  }

}

File: gmall-publisher/src/main/java/com/atguigu/gmall/publisher/controller/PublisherController.java
package com.atguigu.gmall.publisher.controller;


import com.atguigu.gmall.publisher.bean.ByHourText;
import com.atguigu.gmall.publisher.service.PublisherService;
import org.apache.commons.lang.StringUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

@RestController
public class PublisherController {

    @Autowired
    private PublisherService publisherService;

    @GetMapping("realtime-total")
    public List<Map<String,Object>> getTotal(String date) {
        Long dauCount = publisherService.getDauTotalByDate(date);
        List<Map<String,Object>> list = new LinkedList<>();
        Map<String,Object> map1 = new HashMap<String,Object>();
        map1.put("id","dau");
        map1.put("name","新增日活");
        map1.put("value",dauCount);
        list.add(map1);
        Map<String,Object> map2 = new HashMap<String,Object>();
        map2.put("id","new_mid");
        map2.put("name","新增设备");
        map2.put("value",355);
        list.add(map2);
        Double totalAmountByDate = publisherService.getTotalAmountByDate(date);
        Map<String,Object> map3 = new HashMap<String,Object>();
        map3.put("id","order_amount");
        map3.put("name","新增交易额");
        map3.put("value",totalAmountByDate);
        list.add(map3);
        return list;
    }


    @GetMapping("realtime-hour")
    public ByHourText getAmountHoursByDate(String id, String date) {
        if (!StringUtils.equals(id,"order_amount")){
            return new ByHourText();
        }
        return publisherService.getAmountHoursByDate(date);
    }
}
